Crypto/Alchemy/Interpreter/KeysHints.hs

```diff
@@ -36,14 +38,18 @@ newtype Keys = Keys { unKeys :: [Dynamic] } deriving (Monoid, Show)
 -- | Wrapper for a dynamic list of hints.
 newtype Hints = Hints { unHints :: [Dynamic] } deriving (Monoid, Show)
 
+-- | Type synonym for a standard Keys/Hints accumulator
+type KeysHints v m a = StateT Keys (StateT Hints (ReaderT v m)) a
+
+type KeysNoiseCtx v mon = (Algebraic v, MonadReader v mon, MonadRandom mon, MonadAccumulator Keys mon)
```

Comment:
`KeysAccumulatorCtx` strikes me as a better name for this. The `v` argument is essentially the noise param for generating new keys, and the `mon` accumulates `Keys`.

Reply:
Agreed